src/javascript/estraverse-shim.ts

```diff
@@ -1,14 +1,49 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/**
+ * TODO(usergenic): Immediately deprecate this. This `estraverse-shim` module
+ * exists solely to make the transition from `espree/estree` JavaScript parser
+ * to `babylon/babel` and tooling a gradual/piecemeal affair.  It is intended as
+ * a temporary shim that will eventually be removed once all dependent
+ * scanners/visitors and utility functions which use the old `estraverse` style
+ * methods are converted to `babel-traverse` etc.
+ */
+
 import babelTraverse from 'babel-traverse';
 import {NodePath} from 'babel-traverse';
 import * as babel from 'babel-types';
 import {Visitor, VisitResult} from './estree-visitor';
 
+/**
+ * These enum options mimic the estraverse enums that are returned by their
+ * `enterX`/`leaveX` visitor methods to advise flow of the visitor.
+ */
 export enum VisitorOption {
```

Comment:
IIRC, almost none of our code uses this stuff, so it can be pretty easily removed.

Reply:
Eventually.  We do use a few VisitorOption.Skip returns.